### DISCUSSION TITLE
Is that OPC UA cumulative session count matters ?

### DISCUSSION THREAD

**Discussion by saimanoj567** [FreeOpcUa/python-opcua#1293] [Q&A]
I have few doubts is there any possibility that server may turn off if the cumulative session count exceeds 100 ?

**Comment by AndreasHeine** [comment]
if the max session count is reached and you try to connect you should get an error on connect !?

**Reply by saimanoj567** [reply]
Is there is any way to set maximum count to unlimited ? I have reduced the session time out to 10s. I'm monitoring the current session it's not generating any issues.

**Reply by AndreasHeine** [reply]
not sure if it is wise...

https://github.com/FreeOpcUa/python-opcua/blob/858274721766907148b527916e97446dbbf6af6d/opcua/server/binary_server_asyncio.py#L115

https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_server

> backlog is the maximum number of queued connections passed to listen() (defaults to 100).

**Reply by swamper123** [reply]
From: https://reference.opcfoundation.org/v104/Core/docs/Part4/5.6.2/

**

> A Server application should limit the number of Sessions. To protect against misbehaving Clients and denial of service attacks, the Server shall close the oldest Session that is not activated before reaching the maximum number of supported Sessions.

**

So from spec view, you would deactivate this mechanism, unless you know what you are doing and know what your server can handle.

**Reply by saimanoj567** [reply]
Does the session timeout doesn't applicable for subscriptions ? is there any way to set subscription time to infinite? I read some article setting session time to zero is infinite is that true? what is the best way for real-time practice?

**Reply by AndreasHeine** [reply]
never set sessiontimeout to 0 (a session has to die if nothing happens on the wire) the default from opc foundation is 1 hour.

a subsctiption is not bound to a session it can be transfered to another session (but python opcua does not support thtat tbh many devices does not support thtat) so if you close the session the subscription is lost as well...

**Reply by saimanoj567** [reply]
is there any possibility to handle expections of a subscription while running it on threads?
